### ISSUE TITLE
Problems on selecting payment method. 

### ISSUE THREAD

**Issue by borjadevelop** [OFFLINE-GmbH/oc-mall-plugin#226]
I dont know why on select payment method and submit, the mall enter on a catch in line 121 on PaymentMethodSelector.php when try to do that : 

json_decode(decrypt(session()->get('mall.payment_method.data')), true)

I cant continue because it catch all time. If anyone can help me, thanks. 

( Is base mall installation and trying to do with the mall demo theme ) 

THANKS!

**Comment by tobias-kuendig** [comment]
Can you check your `system.log` file and post the error message including the stack trace here?

**Comment by borjadevelop** [comment]
<details>
<summary>System.log</summary>
<br>
[2019-05-15 13:38:07] development.ERROR: Illuminate\Contracts\Encryption\DecryptException: The payload is invalid. in C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Encryption\Encrypter.php:191
Stack trace:
#0 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Encryption\Encrypter.php(134): Illuminate\Encryption\Encrypter->getJsonPayload(NULL)
#1 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Foundation\helpers.php(370): Illuminate\Encryption\Encrypter->decrypt(NULL)
#2 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\plugins\offline\mall\components\PaymentMethodSelector.php(118): decrypt(NULL)
#3 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\plugins\offline\mall\components\PaymentMethodSelector.php(136): OFFLINE\Mall\Components\PaymentMethodSelector->setData()
#4 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\modules\cms\classes\CmsCompoundObject.php(172): OFFLINE\Mall\Components\PaymentMethodSelector->onRun()
#5 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\modules\cms\classes\Controller.php(500): Cms\Classes\CmsCompoundObject->runComponents()
#6 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\modules\cms\classes\Controller.php(393): Cms\Classes\Controller->execPageCycle()
#7 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\modules\cms\classes\Controller.php(224): Cms\Classes\Controller->runPage(Object(Cms\Classes\Page))
#8 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\modules\cms\classes\CmsController.php(50): Cms\Classes\Controller->run('checkout/paymen...')
#9 [internal function]: Cms\Classes\CmsController->run('checkout/paymen...')
#10 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Controller.php(54): call_user_func_array(Array, Array)
#11 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\ControllerDispatcher.php(45): Illuminate\Routing\Controller->callAction('run', Array)
#12 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Route.php(212): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Cms\Classes\CmsController), 'run')
#13 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Route.php(169): Illuminate\Routing\Route->runController()
#14 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Router.php(658): Illuminate\Routing\Route->run()
#15 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(30): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#16 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\plugins\probalear\traductor\classes\LocaleMiddleware.php(29): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#17 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Probalear\Traductor\Classes\LocaleMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#18 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#19 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Middleware\SubstituteBindings.php(41): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#20 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#21 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\View\Middleware\ShareErrorsFromSession.php(49): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#23 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#24 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#25 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Session\Middleware\StartSession.php(63): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#26 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#27 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#28 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse.php(37): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#29 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#30 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#31 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Cookie\Middleware\EncryptCookies.php(66): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#32 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#33 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#34 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(102): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#35 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Router.php(660): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#36 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Router.php(635): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#37 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Router.php(601): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#38 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\october\rain\src\Router\CoreRouter.php(20): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#39 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(176): October\Rain\Router\CoreRouter->dispatch(Object(Illuminate\Http\Request))
#40 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(30): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#41 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode.php(46): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#42 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(149): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#43 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#44 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(102): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#45 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(151): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#46 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(116): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#47 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\index.php(43): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#48 C:\xampp\htdocs\proyectos\julian\ProyectoBaseMallOCMS\server.php(18): require_once('C:\\xampp\\htdocs...')
#49 {main}  
</details>


There are screenshot when i throw the exception in te catch.

![image](https://user-images.githubusercontent.com/22686755/57778246-ecaf8f80-7723-11e9-93e7-e1e9091295ca.png)

![image](https://user-images.githubusercontent.com/22686755/57778202-d6a1cf00-7723-11e9-99c8-cf3e395aebb2.png)

**Comment by tobias-kuendig** [comment]
Did you make any manual changes to the component? Are you up-to-date?

This exception should be caught already:

```php
        try {
            $paymentData = json_decode(decrypt(session()->get('mall.payment_method.data')), true);
        } catch (DecryptException $e) {
            $paymentData = [];
        }
```

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/components/PaymentMethodSelector.php#L122

**Comment by borjadevelop** [comment]
I only did that change to see what exception i have
![image](https://user-images.githubusercontent.com/22686755/57779806-2e8e0500-7727-11e9-8371-a74793b339a2.png)

**Comment by tobias-kuendig** [comment]
Well if you throw that exception it's no wonder there's an Exception! :grin: 

What are you trying to achieve? What doesn't work? If you remove the `throw` line, is there still an error?

**Comment by borjadevelop** [comment]
<details>
<summary>Expected</summary>
<br>
![image](https://user-images.githubusercontent.com/22686755/57781759-18824380-772b-11e9-9099-a474aa086119.png)
<br>
![image](https://user-images.githubusercontent.com/22686755/57781826-3cde2000-772b-11e9-8361-849c3d189b46.png)
<br>
![image](https://user-images.githubusercontent.com/22686755/57782743-21741480-772d-11e9-9aa2-d405cf96e652.png)


</details>



<details>
<summary>what i get </summary>
<br>
![image](https://user-images.githubusercontent.com/22686755/57781759-18824380-772b-11e9-9099-a474aa086119.png)
<br>
![image](https://user-images.githubusercontent.com/22686755/57781826-3cde2000-772b-11e9-8361-849c3d189b46.png)
<br>
![image](https://user-images.githubusercontent.com/22686755/57782848-49fc0e80-772d-11e9-9536-d723b1f2e264.png)

</details>

Same step all the time

**Comment by tobias-kuendig** [comment]
Can you check your network tools? Is the request sent to the backend when you hit «Choose this method»? Does the response contain an error?

Are there any errors in your JS console?

**Comment by borjadevelop** [comment]
Thanks, solved. 
It is because the scripts.htm in the PaymentMethodSelector component doesn't come with the put script twigs and not recognize $.

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/6869f0d1eee599e2bf4aadb847bf16b7223a2671/components/paymentmethodselector/scripts.htm#L1

Thanks for the help.

**Comment by tobias-kuendig** [comment]
Uh yes, this is missing. Thank you for the report. I fixed the issue in d02ed50
